packages/wdio-sauce-service/src/utils.js

```diff
@@ -0,0 +1,28 @@
+import { isW3C } from '@wdio/utils'
+
+export function makeCapabilityFactory(tunnelIdentifier, options) {
+    return capability => {
+        const isCapW3C = Boolean(
+            isW3C(capability) ||
+            capability['sauce:options'] ||
+            !(capability.platform || capability.version)
+        )
+
+        if (!capability['sauce:options'] && isCapW3C) {
+            capability['sauce:options'] = {}
+        }
+
+        Object.assign(capability, options)
+
+        const sauceOptions = isCapW3C ? capability['sauce:options'] : capability
```

Comment:
Can't we just pass in the capability into this method that we want to modify?

Reply:
this would also remove the necessity of:
```js
        if (!capability['sauce:options'] && isCapW3C) {
            capability['sauce:options'] = {}
        }
```

Reply:
I think the capability we want to modify is being passed in, however it might not have a key `sauce:options`, and if we want to use the legacy Sauce Labs protocol it can't have a `sauce:options` key.

So we have to add an empty object at the key `sauce:options` if it isn't already present, unless we think that the user is trying to use the legacy protocol. I'm not sure of a more succinct way to express that in JavaScript that what's already there.